api-test/UnitTest/Validation/DecimalPrecision.cs

```diff
@@ -0,0 +1,29 @@
+using System.ComponentModel.DataAnnotations;
+
+
+namespace api.Models.Tests
+{
+    public class DecimalPrecisionTests
+    {
+        [Theory]
+        [InlineData(12.3456, 1)]
+        [InlineData(123.456, 2)]
+        [InlineData(1234.5678, 3)]
+        public void DecimalPrecision_WhenInvalidPrecision_ShouldReturnError(decimal value, int precision)
+        {
+            var attribute = new DecimalPrecision(precision);
```

Comment:
Hmm, I understand that you are validating the number of decimals accepted.
`DecimalPrecision_Validate_Decimal_Precision_When_Invalid_Precision_Should_Return_An_Error_Message`